src/app/services/weather-api.service.ts

```diff
@@ -27,4 +25,23 @@ export class WeatherAPIService {
   private requestWeather () {
     return this.http.get<Weather>(API_URL)
   }
+
+  private countByDirectionVal (arr: Array<Data> | undefined, val: string) {
+    let i = 0
+    arr?.map((v) => v.wind10m.direction === val ? i++ : i)
```

Comment:
interesting line, usually map is used for mapping each item of array with some callback and create a new array with results of this callback.
For simple iteration let's use forEach instead -> this method will not return anything, just apply callback
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map


BUT, even interesting, let's use that reduce that we talked about (I'm talking about reduce on Array.propotype.reduce :)

reduce is usually the answer for tasks like "let's aggregate data from array / convert array to object / we need some sum of the items in array"
ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

as a hint: you will have 0 as default value and countByDirectionVal will be returning arr.reduce(), because the result of the reduce will be exactly single value - default value that was a result of iteration through all items in array.

thus, your method will have a look:
```
private countByDirectionVal (arr: Array<Data> | undefined, val: string): **number** {
    return arr.reduce( (yourDefaultValueAfterPrevStep: number, secondParamIsItemOfArray) => {
       // TODO: to make this calculationv.wind10m.direction === val ? i++ : i
       // and return yourDefaultValueAfterPrevStep + 1 or yourDefaultValueAfterPrevStep, because it will be as yourDefaultValueAfterPrevStep on the next iteration
     }, 0 )

```